app/main/ipc-helpers.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * This file contains the functions which make use of Main process APIs with already functioning code in Renderer process.
+ */
+import {Notification, nativeImage} from 'electron';
+
+import fetch from 'node-fetch';
+
+export const showDarwinNotification = async (
+	event: Electron.IpcMainEvent,
+	notificationOptions: Electron.NotificationConstructorOptions,
+	profilePicURL: string
+) => {
+	const profilePic = await getNativeImagefromUrl(profilePicURL);
+	notificationOptions.icon = profilePic;
```

Comment:
Why do we need to go through the `NativeImage` dance? Can we not just set `icon` to the URL?

Reply:
Trying to do that from the beginning but it throws an error while converting

![](https://chat.zulip.org/user_uploads/2/d8/MbnNSErq8rTPP_2nPP6x2SN2/profilepic.png)

Reply:
@andersk the API doesn't support passing the URL directly. That's why we have to go through the `nativeImage` process.
https://www.electronjs.org/docs/api/notification#new-notificationoptions